onlinecasino/src/main/java/org/example/model/GameNeedDealer.java

```diff
@@ -0,0 +1,27 @@
+package org.example.model;
+
+import java.math.BigDecimal;
+import java.util.Set;
+
+public abstract class GameNeedDealer extends Game {
+    protected final Dealer dealer;
+    protected final Set<Experience> neededExperience;
+    public GameNeedDealer(int amountOfPlayers, BigDecimal dailyRevenue, Dealer dealer, Set<Experience> neededExperience) {
+        super(amountOfPlayers, dailyRevenue);
+        this.dealer = dealer;
+        this.neededExperience = neededExperience;
+    }
+    @Override
+    public void run() {
+        if (neededExperience.contains(dealer.getExperience())) {
```

Comment:
If the game has no dealer with the necessary experience then nothing happens in your code. It just doesnt take the salary. But irl the game wouldn't run without the dealer, so the software should notify us about it. Its even better if you dont allow creating the game without the pro dealer so its not counting for a nonexistent bank